### PR TITLE
DBZ-677 Renaming record state SMTs

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.jboss.org/browse/DBZ-677

@jpechane Let me know what you think about the new names. I've left the old in place and deprecated them. The idea being to remove them in 0.11. Best review it commit by commit; the second one aims at unifying most of the options across the two SMTs.

### CODE REVIEW

**Comment by jpechane** [comment]
@gunnarmorling The naming is fine for me but I see one asymetry. MongoDB SMT uses term `Document` and database SMT uses term `Record`. But both refers to class `ExtractNewRecordStateConfigDefinition` that contains `Record` in name.
Also I'd like to see a follow-up issue that would extract not only config definitions but part of shared code too.

**Comment by jpechane** [comment]
@gunnarmorling `ExtractNewRecordStateConfigDefinition` lacks header

File: debezium-core/src/main/java/io/debezium/transforms/ExtractNewRecordStateConfigDefinition.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
+package io.debezium.transforms;
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/transforms/ExtractNewRecordStateConfigDefinition.java]:
```suggestion
/*
 * Copyright Debezium Authors.
 *
 * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 */
package io.debezium.transforms;
```

**Comment by gunnarmorling** [comment]
Agh, of course. Added it.

**Comment by jpechane** [comment]
@gunnarmorling Applied, thanks!

**Pull request merged**